<!DOCTYPE html>
<!-- START: inst/pkgdown/templates/layout.html --><!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title>Data Wrangling and Processing for Genomics (modified for NeSI): Assessing Read Quality</title><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" type="text/css" href="../assets/styles.css"><script src="../assets/scripts.js" type="text/javascript"></script><!-- mathjax --><script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      config: ["MMLorHTML.js"],
      jax: ["input/TeX","input/MathML","output/HTML-CSS","output/NativeMML", "output/PreviewHTML"],
      extensions: ["tex2jax.js","mml2jax.js","MathMenu.js","MathZoom.js", "fast-preview.js", "AssistiveMML.js", "a11y/accessibility-menu.js"],
      TeX: {
        extensions: ["AMSmath.js","AMSsymbols.js","noErrors.js","noUndefined.js"]
      },
      tex2jax: {
        inlineMath: [['\\(', '\\)']],
        displayMath: [ ['$$','$$'], ['\\[', '\\]'] ],
        processEscapes: true
      }
    });
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><!-- Responsive Favicon for The Carpentries --><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="manifest" href="../site.webmanifest"><link rel="mask-icon" href="../safari-pinned-tab.svg" color="#5bbad5"><meta name="msapplication-TileColor" content="#da532c"><meta name="theme-color" content="#ffffff"></head><body>
    <header id="top" class="navbar navbar-expand-md navbar-light bg-white top-nav data"><a class="visually-hidden-focusable skip-link" href="#main-content">Skip to main content</a>
  <div class="container-fluid top-nav-container">
    <div class="col-md-6">
      <div class="large-logo">
        <img alt="Data Carpentry" src="../assets/images/data-logo.svg"></div>
    </div>
    <div class="selector-container">
      
      
      <div class="dropdown">
        <button class="btn btn-secondary dropdown-toggle bordered-button" type="button" id="dropdownMenu1" data-bs-toggle="dropdown" aria-expanded="false">
          <i aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View <i data-feather="chevron-down"></i>
        </button>
        <ul class="dropdown-menu" aria-labelledby="dropdownMenu1"><li><button class="dropdown-item" type="button" onclick="window.location.href='../02-quality-control.html';">Learner View</button></li>
        </ul></div>
    </div>
  </div>
  <hr></header><nav class="navbar navbar-expand-xl navbar-light bg-white bottom-nav data" aria-label="Main Navigation"><div class="container-fluid nav-container">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle Navigation">
      <span class="navbar-toggler-icon"></span>
      <span class="menu-title">Menu</span>
    </button>
    <div class="nav-logo">
      <img class="small-logo" alt="Data Carpentry" src="../assets/images/data-logo-sm.svg"></div>
    <div class="lesson-title-md">
      Data Wrangling and Processing for Genomics (modified for NeSI)
    </div>
    <div class="search-icon-sm">
      <!-- TODO: do not show until we have search
        <i role="img" aria-label="search button" data-feather="search"></i>
      -->
    </div>
    <div class="desktop-nav">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0"><li class="nav-item">
          <span class="lesson-title">
            Data Wrangling and Processing for Genomics (modified for NeSI)
          </span>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/key-points.html">Key Points</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/instructor-notes.html">Instructor Notes</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="../instructor/images.html">Extract All Images</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" id="navbarDropdown" data-bs-toggle="dropdown" aria-expanded="false">
            More <i data-feather="chevron-down"></i>
          </button>
          <ul class="dropdown-menu" aria-labelledby="navbarDropdown"><hr><li><a class="dropdown-item" href="discuss.html">Discussion</a></li><li><a class="dropdown-item" href="reference.html">Glossary</a></li>
          </ul></li>
      </ul></div>
    <form class="d-flex col-md-2 search-form">
      <fieldset disabled><input class="form-control me-2 searchbox" type="search" placeholder="Search" aria-label="Search"><button class="btn btn-outline-success tablet-search-button" type="submit">
          <i class="search-icon" data-feather="search" role="img" aria-label="search button"></i>
        </button>
      </fieldset></form>
  </div><!--/div.container-fluid -->
</nav><div class="col-md-12 mobile-title">
  Data Wrangling and Processing for Genomics (modified for NeSI)
</div>

<aside class="col-md-12 lesson-progress"><div style="width: 6%" class="percentage">
    6%
  </div>
  <div class="progress data">
    <div class="progress-bar data" role="progressbar" style="width: 6%" aria-valuenow="6" aria-label="Lesson Progress" aria-valuemin="0" aria-valuemax="100">
    </div>
  </div>
</aside><div class="container">
      <div class="row">
        <!-- START: inst/pkgdown/templates/navbar.html -->
<div id="sidebar-col" class="col-lg-4">
  <div id="sidebar" class="sidebar">
      <nav aria-labelledby="flush-headingEleven"><button role="button" aria-label="close menu" alt="close menu" aria-expanded="true" aria-controls="sidebar" class="collapse-toggle" data-collapse="Collapse " data-episodes="Episodes ">
          <i class="search-icon" data-feather="x" role="img"></i>
        </button>
        <div class="sidebar-inner">
          <div class="row mobile-row">
            <div class="col">
              <div class="sidenav-view-selector">
                <div class="accordion accordion-flush" id="accordionFlush9">
                  <div class="accordion-item">
                    <h2 class="accordion-header" id="flush-headingNine">
                      <button class="accordion-button collapsed" id="instructor" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseNine" aria-expanded="false" aria-controls="flush-collapseNine">
                        <i id="eye" aria-hidden="true" class="icon" data-feather="eye"></i> Instructor View
                      </button>
                    </h2>
                    <div id="flush-collapseNine" class="accordion-collapse collapse" aria-labelledby="flush-headingNine" data-bs-parent="#accordionFlush2">
                      <div class="accordion-body">
                        <a href="../02-quality-control.html">Learner View</a>
                      </div>
                    </div>
                  </div><!--/div.accordion-item-->
                </div><!--/div.accordion-flush-->
              </div><!--div.sidenav-view-selector -->
            </div><!--/div.col -->
      
            <hr></div><!--/div.mobile-row -->

          <div class="accordion accordion-flush" id="accordionFlush11">
            <div class="accordion-item">

              <button id="chapters" class="accordion-button show" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseEleven" aria-expanded="false" aria-controls="flush-collapseEleven">
                <h2 class="accordion-header chapters" id="flush-headingEleven">
                  EPISODES
                </h2>
              </button>
              <div id="flush-collapseEleven" class="accordion-collapse show collapse" aria-labelledby="flush-headingEleven" data-bs-parent="#accordionFlush11">

                <div class="accordion-body">
                  <div class="accordion accordion-flush" id="accordionFlush1">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading1">
        <a href="index.html">Summary and Schedule</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush2">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading2">
        <a href="01-background.html">1. Background and Metadata</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlushcurrent">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-headingcurrent">
      <button class="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapsecurrent" aria-expanded="true" aria-controls="flush-collapsecurrent">
        <span class="visually-hidden">Current Chapter</span>
        <span class="current-chapter">
        2. Assessing Read Quality
        </span>
      </button>
    </div><!--/div.accordion-header-->
        
    <div id="flush-collapsecurrent" class="accordion-collapse collapse show" aria-labelledby="flush-headingcurrent" data-bs-parent="#accordionFlushcurrent">
      <div class="accordion-body">
        <ul><li><a href="#bioinformatic-workflows">Bioinformatic workflows</a></li>
<li><a href="#starting-with-data">Starting with data</a></li>
<li><a href="#quality-control">Quality control</a></li>
<li><a href="#other-notes-optional">Other notes – optional</a></li>
        </ul></div><!--/div.accordion-body-->
    </div><!--/div.accordion-collapse-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush4">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading4">
        <a href="03-trimming.html">3. Trimming and Filtering</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush5">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading5">
        <a href="04-variant_calling.html">4. Variant Calling Workflow</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

<div class="accordion accordion-flush" id="accordionFlush6">
  <div class="accordion-item">
    <div class="accordion-header" id="flush-heading6">
        <a href="05-automation.html">5. Automating a Variant Calling Workflow</a>
    </div><!--/div.accordion-header-->
        
  </div><!--/div.accordion-item-->
</div><!--/div.accordion-flush-->

                </div>
              </div>
            </div>

            <hr class="half-width"><div class="accordion accordion-flush resources" id="accordionFlush12">
              <div class="accordion-item">
                <h2 class="accordion-header" id="flush-headingTwelve">
                  <button class="accordion-button collapsed" id="resources" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseTwelve" aria-expanded="false" aria-controls="flush-collapseTwelve">
                    RESOURCES
                  </button>
                </h2>
                <div id="flush-collapseTwelve" class="accordion-collapse collapse" aria-labelledby="flush-headingTwelve" data-bs-parent="#accordionFlush12">
                  <div class="accordion-body">
                    <ul><li>
                        <a href="../instructor/key-points.html">Key Points</a>
                      </li>
                      <li>
                        <a href="../instructor/instructor-notes.html">Instructor Notes</a>
                      </li>
                      <li>
                        <a href="../instructor/images.html">Extract All Images</a>
                      </li>
                      <hr><li><a class="dropdown-item" href="discuss.html">Discussion</a></li><li><a class="dropdown-item" href="reference.html">Glossary</a></li>
                    </ul></div>
                </div>
              </div>
            </div>
            <hr class="half-width resources"><a href="../instructor/aio.html">See all in one page</a>
            

            <hr class="d-none d-sm-block d-md-none"><div class="d-grid gap-1">
            
            </div>
          </div><!-- /div.accordion -->
        </div><!-- /div.sidebar-inner -->
      </nav></div><!-- /div.sidebar -->
  </div><!-- /div.sidebar-col -->
<!-- END:   inst/pkgdown/templates/navbar.html-->

        <!-- START: inst/pkgdown/templates/content-instructor.html -->
  <div class="col-xl-8 col-lg-12 primary-content">
    <nav class="lesson-content mx-md-4" aria-label="Previous and Next Chapter"><!-- content for small screens --><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/01-background.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/03-trimming.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/01-background.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Background and
        </a>
        <a class="chapter-link float-end" href="../instructor/03-trimming.html" rel="next">
          Next: Trimming and... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
      <hr></nav><main id="main-content" class="main-content"><div class="container lesson-content">
        <h1>Assessing Read Quality</h1>
        <p>Last updated on 2023-11-27 |
        
        <a href="https://github.com/otagobioinformaticsspringschool/wrangling-genomics-nesi/edit/main/episodes/02-quality-control.md" class="external-link">Edit this page <i aria-hidden="true" data-feather="edit"></i></a></p>
        
        
        
        <p>Estimated time: <i aria-hidden="true" data-feather="clock"></i> 50 minutes</p>
        
        <div class="text-end">
          <button role="button" aria-pressed="false" tabindex="0" id="expand-code" class="pull-right" data-expand="Expand All Solutions " data-collapse="Collapse All Solutions "> Expand All Solutions <i aria-hidden="true" data-feather="plus"></i></button>
        </div>

        

<div class="overview card">
<h2 class="card-header">Overview</h2>
<div class="row g-0">
<div class="col-md-4">
<div class="card-body">
<div class="inner">
<h3 class="card-title">Questions</h3>
<ul><li>How can I describe the quality of my data?</li>
</ul></div>
</div>
</div>
<div class="col-md-8">
<div class="card-body">
<div class="inner bordered">
<h3 class="card-title">Objectives</h3>
<ul><li>Explain how a FASTQ file encodes per-base quality scores.</li>
<li>Interpret a FastQC plot summarizing per-base quality across all
reads.</li>
<li>Use <code>for</code> loops to automate operations on multiple
files.</li>
</ul></div>
</div>
</div>
</div>
</div>
<p><strong>This lesson has been adapted from the original <a href="https://datacarpentry.org/wrangling-genomics/" class="external-link">Data Carpentry -
Wrangling Genomics</a> to be run using the NeSI infrastructure as part
of the Otago Bioinformatics Spring School instead of AWS.</strong></p>
<section id="bioinformatic-workflows"><h2 class="section-heading">Bioinformatic workflows<a class="anchor" aria-label="anchor" href="#bioinformatic-workflows"></a>
</h2>
<hr class="half-width"><p>When working with high-throughput sequencing data, the raw reads you
get off of the sequencer will need to pass through a number of different
tools in order to generate your final desired output. The execution of
this set of tools in a specified order is commonly referred to as a
<em>workflow</em> or a <em>pipeline</em>.</p>
<p>An example of the workflow we will be using for our variant calling
analysis is provided below with a brief description of each step.</p>
<figure><img src="../fig/variant_calling_workflow.png" alt="workflow" class="figure mx-auto d-block"></figure><ol style="list-style-type: decimal"><li>Quality control - Assessing quality using FastQC</li>
<li>Quality control - Trimming and/or filtering reads (if
necessary)</li>
<li>Align reads to reference genome</li>
<li>Perform post-alignment clean-up</li>
<li>Variant calling</li>
</ol><p>These workflows in bioinformatics adopt a plug-and-play approach in
that the output of one tool can be easily used as input to another tool
without any extensive configuration. Having standards for data formats
is what makes this feasible. Standards ensure that data is stored in a
way that is generally accepted and agreed upon within the community. The
tools that are used to analyze data at different stages of the workflow
are therefore built under the assumption that the data will be provided
in a specific format.</p>
</section><section id="starting-with-data"><h2 class="section-heading">Starting with data<a class="anchor" aria-label="anchor" href="#starting-with-data"></a>
</h2>
<hr class="half-width"><p>Often times, the first step in a bioinformatic workflow is getting
the data you want to work with onto a computer where you can work with
it. If you have outsourced sequencing of your data, the sequencing
center will usually provide you with a link that you can use to download
your data. Today we will be working with publicly available sequencing
data.</p>
<p>We are studying a population of <em>Escherichia coli</em> (designated
Ara-3), which were propagated for more than 50,000 generations in a
glucose-limited minimal medium. We will be working with three samples
from this experiment, one from 5,000 generations, one from 15,000
generations, and one from 50,000 generations. The population changed
substantially during the course of the experiment, and we will be
exploring how with our variant calling workflow.</p>
<p>The data are paired-end, so we will download two files for each
sample. We will use the <a href="https://www.ebi.ac.uk/ena" class="external-link">European
Nucleotide Archive</a> to get our data. The ENA “provides a
comprehensive record of the world’s nucleotide sequencing information,
covering raw sequencing data, sequence assembly information and
functional annotation.” The ENA also provides sequencing data in the
fastq format, an important format for sequencing reads that we will be
learning about today.</p>
<p>To save time, the data has already been downloaded for you and placed
in <code>~/obss_2023/genomic_dna/data/untrimmed_fastq</code>.</p>
<p>Today we’re going to be working inside of
<code>~/obss_2023/genomic_dna</code>. The main directory structure for
<code>genomic_dna/</code> (inside of <code>~/obss_2023/</code>)
currently looks like this currently</p>
<div class="codewrapper sourceCode" id="cb1">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">genomic_dna/</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>    <span class="kw">|</span><span class="ex">-adapters/</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>    <span class="kw">|</span><span class="ex">-data/</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>        <span class="kw">`</span><span class="ex">-untrimmed_fastq/</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>    <span class="kw">|</span><span class="ex">-docs/</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>    <span class="kw">`</span><span class="at">-results</span>/</span></code></pre>
</div>
<p>Now lets navigate to where the untrimmed fastq data is stored.</p>
<div class="codewrapper sourceCode" id="cb2">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="bu">cd</span> ~/obss_2023/genomic_dna/data/untrimmed_fastq</span></code></pre>
</div>
<p>To download the data, run the commands below.</p>
<p>Here we are using the <code>-p</code> option for <code>mkdir</code>.
This option allows <code>mkdir</code> to create the new directory, even
if one of the parent directories does not already exist. It also
supresses errors if the directory already exists, without overwriting
that directory.</p>
<p>It will take about 15 minutes to download the files.</p>
<div id="original-way-to-obtain-the-data-use-if-not-on-nesi" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="original-way-to-obtain-the-data-use-if-not-on-nesi" class="callout-inner">
<h3 class="callout-title">Original way to obtain the data (Use if not on
NeSI)<a class="anchor" aria-label="anchor" href="#original-way-to-obtain-the-data-use-if-not-on-nesi"></a>
</h3>
<div class="callout-content">
<p>To download the data, run the commands below.</p>
<p>Here we are using the <code>-p</code> option for <code>mkdir</code>.
This option allows <code>mkdir</code> to create the new directory, even
if one of the parent directories doesn’t already exist. It also
supresses errors if the directory already exists, without overwriting
that directory.</p>
<p>It will take about 15 minutes to download the files.</p>
<div class="codewrapper sourceCode" id="cb3">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> ~/obss_2023/data/untrimmed_fastq/</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="bu">cd</span> ~/obss_2023/data/untrimmed_fastq</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/004/SRR2589044/SRR2589044_1.fastq.gz</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/004/SRR2589044/SRR2589044_2.fastq.gz</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/003/SRR2584863/SRR2584863_1.fastq.gz</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/003/SRR2584863/SRR2584863_2.fastq.gz</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/006/SRR2584866/SRR2584866_1.fastq.gz</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-O</span> ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR258/006/SRR2584866/SRR2584866_2.fastq.gz</span></code></pre>
</div>
</div>
</div>
</div>
<p>The data comes in a compressed format, which is why there is a
<code>.gz</code> at the end of the file names. This makes it faster to
transfer, and allows it to take up less space on our computer. Let’s
unzip one of the files so that we can look at the fastq format.</p>
<div class="codewrapper sourceCode" id="cb4">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="ex">$</span> gunzip SRR2584863_1.fastq.gz</span></code></pre>
</div>
</section><section id="quality-control"><h2 class="section-heading">Quality control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h2>
<hr class="half-width"><p>We will now assess the quality of the sequence reads contained in our
fastq files.</p>
<figure><img src="../fig/var_calling_workflow_qc.png" alt="workflow_qc" class="figure mx-auto d-block"></figure><div class="section level3">
<h3 id="details-on-the-fastq-format">Details on the FASTQ format<a class="anchor" aria-label="anchor" href="#details-on-the-fastq-format"></a></h3>
<p>Although it looks complicated (and it is), we can understand the <a href="https://en.wikipedia.org/wiki/FASTQ_format" class="external-link">fastq</a> format with
a little decoding. Some rules about the format include…</p>
<table class="table"><colgroup><col width="3%"><col width="96%"></colgroup><thead><tr class="header"><th>Line</th>
<th>Description</th>
</tr></thead><tbody><tr class="odd"><td>1</td>
<td>Always begins with ‘@’ and then information about the read</td>
</tr><tr class="even"><td>2</td>
<td>The actual DNA sequence</td>
</tr><tr class="odd"><td>3</td>
<td>Always begins with a ‘+’ and sometimes the same info in line 1</td>
</tr><tr class="even"><td>4</td>
<td>Has a string of characters which represent the quality scores; must
have same number of characters as line 2</td>
</tr></tbody></table><p>We can view the first complete read in one of the files our dataset
by using <code>head</code> to look at the first four lines.</p>
<div class="codewrapper sourceCode" id="cb5">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">$</span> head <span class="at">-n</span> 4 SRR2584863_1.fastq</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>@SRR2584863.1 HWI-ST957:244:H73TDADXX:1:1101:4712:2181/1
TTCACATCCTGACCATTCAGTTGAGCAAAATAGTTCTTCAGTGCCTGTTTAACCGAGTCACGCAGGGGTTTTTGGGTTACCTGATCCTGAGAGTTAACGGTAGAAACGGTCAGTACGTCAGAATTTACGCGTTGTTCGAACATAGTTCTG
+
CCCFFFFFGHHHHJIJJJJIJJJIIJJJJIIIJJGFIIIJEDDFEGGJIFHHJIJJDECCGGEGIIJFHFFFACD:BBBDDACCCCAA@@CA@C&gt;C3&gt;@5(8&amp;&gt;C:9?8+89&lt;4(:83825C(:A#########################</code></pre>
</div>
<p>Line 4 shows the quality for each nucleotide in the read. Quality is
interpreted as the probability of an incorrect base call (e.g. 1 in 10)
or, equivalently, the base call accuracy (e.g. 90%). To make it possible
to line up each individual nucleotide with its quality score, the
numerical score is converted into a code where each individual character
represents the numerical quality score for an individual nucleotide. For
example, in the line above, the quality score line is:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>CCCFFFFFGHHHHJIJJJJIJJJIIJJJJIIIJJGFIIIJEDDFEGGJIFHHJIJJDECCGGEGIIJFHFFFACD:BBBDDACCCCAA@@CA@C&gt;C3&gt;@5(8&amp;&gt;C:9?8+89&lt;4(:83825C(:A#########################</code></pre>
</div>
<p>The numerical value assigned to each of these characters depends on
the sequencing platform that generated the reads. The sequencing machine
used to generate our data uses the standard Sanger quality PHRED score
encoding, using Illumina version 1.8 onwards. Each character is assigned
a quality score between 0 and 41 as shown in the chart below.</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Quality encoding: !"#$%&amp;'()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJ
                   |         |         |         |         |
Quality score:    01........11........21........31........41</code></pre>
</div>
<p>Each quality score represents the probability that the corresponding
nucleotide call is incorrect. This quality score is logarithmically
based, so a quality score of 10 reflects a base call accuracy of 90%,
but a quality score of 20 reflects a base call accuracy of 99%. These
probability values are the results from the base calling algorithm and
depend on how much signal was captured for the base incorporation.</p>
<p>Looking back at our read:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>@SRR2584863.1 HWI-ST957:244:H73TDADXX:1:1101:4712:2181/1
TTCACATCCTGACCATTCAGTTGAGCAAAATAGTTCTTCAGTGCCTGTTTAACCGAGTCACGCAGGGGTTTTTGGGTTACCTGATCCTGAGAGTTAACGGTAGAAACGGTCAGTACGTCAGAATTTACGCGTTGTTCGAACATAGTTCTG
+
CCCFFFFFGHHHHJIJJJJIJJJIIJJJJIIIJJGFIIIJEDDFEGGJIFHHJIJJDECCGGEGIIJFHFFFACD:BBBDDACCCCAA@@CA@C&gt;C3&gt;@5(8&amp;&gt;C:9?8+89&lt;4(:83825C(:A#########################</code></pre>
</div>
<p>we can now see that there is a range of quality scores, but that the
end of the sequence is very poor (<code>#</code> = a quality score of
2).</p>
<div id="exercise" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise" class="callout-inner">
<h3 class="callout-title">Exercise<a class="anchor" aria-label="anchor" href="#exercise"></a>
</h3>
<div class="callout-content">
<p>What is the last read in the <code>SRR2584863_1.fastq</code> file?
How confident are you in this read?</p>
</div>
</div>
</div>
<div id="accordionSolution1" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution1" aria-expanded="false" aria-controls="collapseSolution1">
  <h4 class="accordion-header" id="headingSolution1">Show me the solution</h4>
</button>
<div id="collapseSolution1" class="accordion-collapse collapse" data-bs-parent="#accordionSolution1" aria-labelledby="headingSolution1">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb10">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="ex">$</span> tail <span class="at">-n</span> 4 SRR2584863_1.fastq</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>@SRR2584863.1553259 HWI-ST957:245:H73R4ADXX:2:2216:21048:100894/1
CTGCAATACCACGCTGATCTTTCACATGATGTAAGAAAAGTGGGATCAGCAAACCGGGTGCTGCTGTGGCTAGTTGCAGCAAACCATGCAGTGAACCCGCCTGTGCTTCGCTATAGCCGTGACTGATGAGGATCGCCGGAAGCCAGCCAA
+
CCCFFFFFHHHHGJJJJJJJJJHGIJJJIJJJJIJJJJIIIIJJJJJJJJJJJJJIIJJJHHHHHFFFFFEEEEEDDDDDDDDDDDDDDDDDCDEDDBDBDDBDDDDDDDDDBDEEDDDD7@BDDDDDD&gt;AA&gt;?B?&lt;@BDD@BDC?BDA?</code></pre>
</div>
<p>This read has more consistent quality at its end than the first read
that we looked at, but still has a range of quality scores, most of them
high. We will look at variations in position-based quality in just a
moment.</p>
</div>
</div>
</div>
</div>
<p>At this point, lets validate that all the relevant tools are
installed. If you are using the AWS AMI then these <em>should</em> be
preinstalled.</p>
<div id="callout2" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Callout<a class="anchor" aria-label="anchor" href="#callout2"></a>
</h3>
<div class="callout-content">
<p>On NeSI, we load the tools by running a script that has the needed
software and versions specified.</p>
<div class="codewrapper sourceCode" id="cb12">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="ex">$</span> source ~/obss_2023/genomic_dna/modload.sh</span></code></pre>
</div>
<div class="section level2">
<h2 id="modules">Modules<a class="anchor" aria-label="anchor" href="#modules"></a></h2>
<p>On NeSI, software is managed through <code>modules</code> which
enable many versions of the same program to co-exist. This particular
<code>modload.sh</code> script contains the commands that loads the
specific software modules needed for our analysis. It uses the
<code>module load</code> command which is then followed by the module
you wish to use. In the case of <code>fastqc</code> we use
<code>module load FastQC</code>. To find the modules for software
managed on NeSI you can use
<code>module spider &lt;program name&gt;</code>,
e.g. <code>module spider fastqc</code> will return the names of the
modules related to <code>fastqc</code>.</p>
<p>For more information about using modules on NeSI see their <a href="https://support.nesi.org.nz/hc/en-gb/articles/360000360576" class="external-link">documentation</a>.</p>
</div>
</div>
</div>
</div>
<div class="codewrapper sourceCode" id="cb13">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="ex">$</span> fastqc <span class="at">-h</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>            <span class="ex">FastQC</span> <span class="at">-</span> A high throughput sequence QC analysis tool</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="ex">SYNOPSIS</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>        <span class="ex">fastqc</span> seqfile1 seqfile2 .. seqfileN</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>    <span class="ex">fastqc</span> [-o output dir] [--<span class="er">(</span><span class="ex">no</span><span class="kw">)</span><span class="ex">extract]</span> [-f fastq<span class="kw">|</span><span class="ex">bam</span><span class="kw">|</span><span class="ex">sam]</span></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>           <span class="ex">[-c</span> contaminant file] seqfile1 .. seqfileN</span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" tabindex="-1"></a><span class="ex">DESCRIPTION</span></span>
<span id="cb13-12"><a href="#cb13-12" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" tabindex="-1"></a>    <span class="ex">FastQC</span> reads a set of sequence files and produces from each one a quality</span>
<span id="cb13-14"><a href="#cb13-14" tabindex="-1"></a>    <span class="ex">control</span> report consisting of a number of different modules, each one of</span>
<span id="cb13-15"><a href="#cb13-15" tabindex="-1"></a>    <span class="fu">which</span> will help to identify a different potential type of problem in your</span>
<span id="cb13-16"><a href="#cb13-16" tabindex="-1"></a>    <span class="ex">data.</span></span>
<span id="cb13-17"><a href="#cb13-17" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" tabindex="-1"></a>    <span class="ex">If</span> no files to process are specified on the command line then the program</span>
<span id="cb13-19"><a href="#cb13-19" tabindex="-1"></a>    <span class="ex">will</span> start as an interactive graphical application.  If files are provided</span>
<span id="cb13-20"><a href="#cb13-20" tabindex="-1"></a>    <span class="ex">on</span> the command line then the program will run with no user interaction</span>
<span id="cb13-21"><a href="#cb13-21" tabindex="-1"></a>    <span class="ex">required.</span>  In this mode it is suitable for inclusion into a standardised</span>
<span id="cb13-22"><a href="#cb13-22" tabindex="-1"></a>    <span class="ex">analysis</span> pipeline.</span>
<span id="cb13-23"><a href="#cb13-23" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" tabindex="-1"></a>    <span class="ex">The</span> options for the program as as follows:</span>
<span id="cb13-25"><a href="#cb13-25" tabindex="-1"></a></span>
<span id="cb13-26"><a href="#cb13-26" tabindex="-1"></a>    <span class="ex">-h</span> <span class="at">--help</span>       Print this help file and exit</span>
<span id="cb13-27"><a href="#cb13-27" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" tabindex="-1"></a>    <span class="ex">-v</span> <span class="at">--version</span>    Print the version of the program and exit</span>
<span id="cb13-29"><a href="#cb13-29" tabindex="-1"></a></span>
<span id="cb13-30"><a href="#cb13-30" tabindex="-1"></a>    <span class="ex">-o</span> <span class="at">--outdir</span>     Create all output files in the specified output directory.</span>
<span id="cb13-31"><a href="#cb13-31" tabindex="-1"></a>                    <span class="ex">Please</span> note that this directory must exist as the program</span>
<span id="cb13-32"><a href="#cb13-32" tabindex="-1"></a>                    <span class="ex">will</span> not create it.  If this option is not set then the</span>
<span id="cb13-33"><a href="#cb13-33" tabindex="-1"></a>                    <span class="ex">output</span> file for each sequence file is created in the same</span>
<span id="cb13-34"><a href="#cb13-34" tabindex="-1"></a>                    <span class="ex">directory</span> as the sequence file which was processed.</span>
<span id="cb13-35"><a href="#cb13-35" tabindex="-1"></a></span>
<span id="cb13-36"><a href="#cb13-36" tabindex="-1"></a>    <span class="ex">--casava</span>        Files come from raw casava output. Files in the same sample</span>
<span id="cb13-37"><a href="#cb13-37" tabindex="-1"></a>                    <span class="ex">group</span> <span class="er">(</span><span class="ex">differing</span> only by the group number<span class="kw">)</span> <span class="ex">will</span> be analysed</span>
<span id="cb13-38"><a href="#cb13-38" tabindex="-1"></a>                    <span class="fu">as</span> a set rather than individually. Sequences with the filter</span>
<span id="cb13-39"><a href="#cb13-39" tabindex="-1"></a>                    <span class="ex">flag</span> set in the header will be excluded from the analysis.</span>
<span id="cb13-40"><a href="#cb13-40" tabindex="-1"></a>                    <span class="ex">Files</span> must have the same names given to them by casava</span>
<span id="cb13-41"><a href="#cb13-41" tabindex="-1"></a>                    <span class="kw">(</span><span class="ex">including</span> being gzipped and ending with .gz<span class="kw">)</span> <span class="ex">otherwise</span> they</span>
<span id="cb13-42"><a href="#cb13-42" tabindex="-1"></a>                    <span class="ex">will</span> not be grouped together correctly.</span>
<span id="cb13-43"><a href="#cb13-43" tabindex="-1"></a></span>
<span id="cb13-44"><a href="#cb13-44" tabindex="-1"></a>    <span class="ex">--nano</span>          Files come from naopore sequences and are in fast5 format. In</span>
<span id="cb13-45"><a href="#cb13-45" tabindex="-1"></a>                    <span class="ex">this</span> mode you can pass in directories to process and the program</span>
<span id="cb13-46"><a href="#cb13-46" tabindex="-1"></a>                    <span class="ex">will</span> take in all fast5 files within those directories and produce</span>
<span id="cb13-47"><a href="#cb13-47" tabindex="-1"></a>                    <span class="ex">a</span> single output file from the sequences found in all files.</span>
<span id="cb13-48"><a href="#cb13-48" tabindex="-1"></a></span>
<span id="cb13-49"><a href="#cb13-49" tabindex="-1"></a>    <span class="ex">--nofilter</span>      If running with <span class="at">--casava</span> then don<span class="st">'t remove read flagged by</span></span>
<span id="cb13-50"><a href="#cb13-50" tabindex="-1"></a><span class="st">                    casava as poor quality when performing the QC analysis.</span></span>
<span id="cb13-51"><a href="#cb13-51" tabindex="-1"></a></span>
<span id="cb13-52"><a href="#cb13-52" tabindex="-1"></a><span class="st">    --extract       If set then the zipped output file will be uncompressed in</span></span>
<span id="cb13-53"><a href="#cb13-53" tabindex="-1"></a><span class="st">                    the same directory after it has been created.  By default</span></span>
<span id="cb13-54"><a href="#cb13-54" tabindex="-1"></a><span class="st">                    this option will be set if fastqc is run in non-interactive</span></span>
<span id="cb13-55"><a href="#cb13-55" tabindex="-1"></a><span class="st">                    mode.</span></span>
<span id="cb13-56"><a href="#cb13-56" tabindex="-1"></a></span>
<span id="cb13-57"><a href="#cb13-57" tabindex="-1"></a><span class="st">    -j --java       Provides the full path to the java binary you want to use to</span></span>
<span id="cb13-58"><a href="#cb13-58" tabindex="-1"></a><span class="st">                    launch fastqc. If not supplied then java is assumed to be in</span></span>
<span id="cb13-59"><a href="#cb13-59" tabindex="-1"></a><span class="st">                    your path.</span></span>
<span id="cb13-60"><a href="#cb13-60" tabindex="-1"></a></span>
<span id="cb13-61"><a href="#cb13-61" tabindex="-1"></a><span class="st">    --noextract     Do not uncompress the output file after creating it.  You</span></span>
<span id="cb13-62"><a href="#cb13-62" tabindex="-1"></a><span class="st">                    should set this option if you do not wish to uncompress</span></span>
<span id="cb13-63"><a href="#cb13-63" tabindex="-1"></a><span class="st">                    the output when running in non-interactive mode.</span></span>
<span id="cb13-64"><a href="#cb13-64" tabindex="-1"></a></span>
<span id="cb13-65"><a href="#cb13-65" tabindex="-1"></a><span class="st">    --nogroup       Disable grouping of bases for reads &gt;50bp. All reports will</span></span>
<span id="cb13-66"><a href="#cb13-66" tabindex="-1"></a><span class="st">                    show data for every base in the read.  WARNING: Using this</span></span>
<span id="cb13-67"><a href="#cb13-67" tabindex="-1"></a><span class="st">                    option will cause fastqc to crash and burn if you use it on</span></span>
<span id="cb13-68"><a href="#cb13-68" tabindex="-1"></a><span class="st">                    really long reads, and your plots may end up a ridiculous size.</span></span>
<span id="cb13-69"><a href="#cb13-69" tabindex="-1"></a><span class="st">                    You have been warned!</span></span>
<span id="cb13-70"><a href="#cb13-70" tabindex="-1"></a></span>
<span id="cb13-71"><a href="#cb13-71" tabindex="-1"></a><span class="st">    -f --format     Bypasses the normal sequence file format detection and</span></span>
<span id="cb13-72"><a href="#cb13-72" tabindex="-1"></a><span class="st">                    forces the program to use the specified format.  Valid</span></span>
<span id="cb13-73"><a href="#cb13-73" tabindex="-1"></a><span class="st">                    formats are bam,sam,bam_mapped,sam_mapped and fastq</span></span>
<span id="cb13-74"><a href="#cb13-74" tabindex="-1"></a></span>
<span id="cb13-75"><a href="#cb13-75" tabindex="-1"></a><span class="st">    -t --threads    Specifies the number of files which can be processed</span></span>
<span id="cb13-76"><a href="#cb13-76" tabindex="-1"></a><span class="st">                    simultaneously.  Each thread will be allocated 250MB of</span></span>
<span id="cb13-77"><a href="#cb13-77" tabindex="-1"></a><span class="st">                    memory so you shouldn'</span>t run more threads than your</span>
<span id="cb13-78"><a href="#cb13-78" tabindex="-1"></a>                    <span class="ex">available</span> memory will cope with, and not more than</span>
<span id="cb13-79"><a href="#cb13-79" tabindex="-1"></a>                    <span class="ex">6</span> threads on a 32 bit machine</span>
<span id="cb13-80"><a href="#cb13-80" tabindex="-1"></a></span>
<span id="cb13-81"><a href="#cb13-81" tabindex="-1"></a>    <span class="ex">-c</span>              Specifies a non-default file which contains the list of</span>
<span id="cb13-82"><a href="#cb13-82" tabindex="-1"></a>    <span class="ex">--contaminants</span>  contaminants to screen overrepresented sequences against.</span>
<span id="cb13-83"><a href="#cb13-83" tabindex="-1"></a>                    <span class="ex">The</span> file must contain sets of named contaminants in the</span>
<span id="cb13-84"><a href="#cb13-84" tabindex="-1"></a>                    <span class="ex">form</span> name<span class="pp">[</span><span class="ss">tab</span><span class="pp">]</span>sequence.  Lines prefixed with a hash will</span>
<span id="cb13-85"><a href="#cb13-85" tabindex="-1"></a>                    <span class="ex">be</span> ignored.</span>
<span id="cb13-86"><a href="#cb13-86" tabindex="-1"></a></span>
<span id="cb13-87"><a href="#cb13-87" tabindex="-1"></a>    <span class="ex">-a</span>              Specifies a non-default file which contains the list of</span>
<span id="cb13-88"><a href="#cb13-88" tabindex="-1"></a>    <span class="ex">--adapters</span>      adapter sequences which will be explicity searched against</span>
<span id="cb13-89"><a href="#cb13-89" tabindex="-1"></a>                    <span class="ex">the</span> library. The file must contain sets of named adapters</span>
<span id="cb13-90"><a href="#cb13-90" tabindex="-1"></a>                    <span class="er">in</span> <span class="ex">the</span> form name<span class="pp">[</span><span class="ss">tab</span><span class="pp">]</span>sequence.  Lines prefixed with a hash</span>
<span id="cb13-91"><a href="#cb13-91" tabindex="-1"></a>                    <span class="ex">will</span> be ignored.</span>
<span id="cb13-92"><a href="#cb13-92" tabindex="-1"></a></span>
<span id="cb13-93"><a href="#cb13-93" tabindex="-1"></a>    <span class="ex">-l</span>              Specifies a non-default file which contains a set of criteria</span>
<span id="cb13-94"><a href="#cb13-94" tabindex="-1"></a>    <span class="ex">--limits</span>        which will be used to determine the warn/error limits for the</span>
<span id="cb13-95"><a href="#cb13-95" tabindex="-1"></a>                    <span class="ex">various</span> modules.  This file can also be used to selectively</span>
<span id="cb13-96"><a href="#cb13-96" tabindex="-1"></a>                    <span class="ex">remove</span> some modules from the output all together.  The format</span>
<span id="cb13-97"><a href="#cb13-97" tabindex="-1"></a>                    <span class="ex">needs</span> to mirror the default limits.txt file found in the</span>
<span id="cb13-98"><a href="#cb13-98" tabindex="-1"></a>                    <span class="ex">Configuration</span> folder.</span>
<span id="cb13-99"><a href="#cb13-99" tabindex="-1"></a></span>
<span id="cb13-100"><a href="#cb13-100" tabindex="-1"></a>   <span class="ex">-k</span> <span class="at">--kmers</span>       Specifies the length of Kmer to look for in the Kmer content</span>
<span id="cb13-101"><a href="#cb13-101" tabindex="-1"></a>                    <span class="ex">module.</span> Specified Kmer length must be between 2 and 10. Default</span>
<span id="cb13-102"><a href="#cb13-102" tabindex="-1"></a>                    <span class="ex">length</span> is 7 if not specified.</span>
<span id="cb13-103"><a href="#cb13-103" tabindex="-1"></a></span>
<span id="cb13-104"><a href="#cb13-104" tabindex="-1"></a>   <span class="ex">-q</span> <span class="at">--quiet</span>       Supress all progress messages on stdout and only report errors.</span>
<span id="cb13-105"><a href="#cb13-105" tabindex="-1"></a></span>
<span id="cb13-106"><a href="#cb13-106" tabindex="-1"></a>   <span class="ex">-d</span> <span class="at">--dir</span>         Selects a directory to be used for temporary files written when</span>
<span id="cb13-107"><a href="#cb13-107" tabindex="-1"></a>                    <span class="ex">generating</span> report images. Defaults to system temp directory if</span>
<span id="cb13-108"><a href="#cb13-108" tabindex="-1"></a>                    <span class="ex">not</span> specified.</span>
<span id="cb13-109"><a href="#cb13-109" tabindex="-1"></a></span>
<span id="cb13-110"><a href="#cb13-110" tabindex="-1"></a><span class="ex">BUGS</span></span>
<span id="cb13-111"><a href="#cb13-111" tabindex="-1"></a></span>
<span id="cb13-112"><a href="#cb13-112" tabindex="-1"></a>    <span class="ex">Any</span> bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk</span>
<span id="cb13-113"><a href="#cb13-113" tabindex="-1"></a>    <span class="ex">or</span> in www.bioinformatics.babraham.ac.uk/bugzilla/</span></code></pre>
</div>
<p>if fastqc is not installed then you would expect to see an error
like</p>
<pre><code>$ fastqc -h
The program 'fastqc' is currently not installed. You can install it by typing:
sudo apt-get install fastqc</code></pre>
<p>If this happens check with your instructor before trying to install
it.</p>
</div>
<div class="section level3">
<h3 id="assessing-quality-using-fastqc">Assessing quality using FastQC<a class="anchor" aria-label="anchor" href="#assessing-quality-using-fastqc"></a></h3>
<p>In real life, you will not be assessing the quality of your reads by
visually inspecting your FASTQ files. Rather, you will be using a
software program to assess read quality and filter out poor quality
reads. We will first use a program called <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" class="external-link">FastQC</a>
to visualize the quality of our reads. Later in our workflow, we will
use another program to filter out poor quality reads.</p>
<p>FastQC has a number of features which can give you a quick impression
of any problems your data may have, so you can take these issues into
consideration before moving forward with your analyses. Rather than
looking at quality scores for each individual read, FastQC looks at
quality collectively across all reads within a sample. The image below
shows one FastQC-generated plot that indicates a very high quality
sample:</p>
<figure><img src="../fig/good_quality1.8.png" alt="good_quality" class="figure mx-auto d-block"></figure><p>The x-axis displays the base position in the read, and the y-axis
shows quality scores. In this example, the sample contains reads that
are 40 bp long. This is much shorter than the reads we are working with
in our workflow. For each position, there is a box-and-whisker plot
showing the distribution of quality scores for all reads at that
position. The horizontal red line indicates the median quality score and
the yellow box shows the 1st to 3rd quartile range. This means that 50%
of reads have a quality score that falls within the range of the yellow
box at that position. The whiskers show the absolute range, which covers
the lowest (0th quartile) to highest (4th quartile) values.</p>
<p>For each position in this sample, the quality values do not drop much
lower than 32. This is a high quality score. The plot background is also
color-coded to identify good (green), acceptable (yellow), and bad (red)
quality scores.</p>
<p>Now let’s take a look at a quality plot on the other end of the
spectrum.</p>
<figure><img src="../fig/bad_quality1.8.png" alt="bad_quality" class="figure mx-auto d-block"></figure><p>Here, we see positions within the read in which the boxes span a much
wider range. Also, quality scores drop quite low into the “bad” range,
particularly on the tail end of the reads. The FastQC tool produces
several other diagnostic plots to assess sample quality, in addition to
the one plotted above.</p>
</div>
<div class="section level3">
<h3 id="running-fastqc">Running FastQC<a class="anchor" aria-label="anchor" href="#running-fastqc"></a></h3>
<p>We will now assess the quality of the reads that we downloaded.
First, make sure you are still in the <code>untrimmed_fastq</code>
directory</p>
<div class="codewrapper sourceCode" id="cb15">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="ex">$</span> cd ~/obss_2023/genomic_dna/data/untrimmed_fastq/</span></code></pre>
</div>
<div id="exercise-1" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-1" class="callout-inner">
<h3 class="callout-title">Exercise<a class="anchor" aria-label="anchor" href="#exercise-1"></a>
</h3>
<div class="callout-content">
<p>How big are the files? (Hint: Look at the options for the
<code>ls</code> command to see how to show file sizes.)</p>
</div>
</div>
</div>
<div id="accordionSolution2" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution2" aria-expanded="false" aria-controls="collapseSolution2">
  <h4 class="accordion-header" id="headingSolution2">Show me the solution</h4>
</button>
<div id="collapseSolution2" class="accordion-collapse collapse" data-bs-parent="#accordionSolution2" aria-labelledby="headingSolution2">
<div class="accordion-body">
<div class="codewrapper sourceCode" id="cb16">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-l</span> <span class="at">-h</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>-rw-rw-r-- 1 dcuser dcuser 545M Jul  6 20:27 SRR2584863_1.fastq
-rw-rw-r-- 1 dcuser dcuser 183M Jul  6 20:29 SRR2584863_2.fastq.gz
-rw-rw-r-- 1 dcuser dcuser 309M Jul  6 20:34 SRR2584866_1.fastq.gz
-rw-rw-r-- 1 dcuser dcuser 296M Jul  6 20:37 SRR2584866_2.fastq.gz
-rw-rw-r-- 1 dcuser dcuser 124M Jul  6 20:22 SRR2589044_1.fastq.gz
-rw-rw-r-- 1 dcuser dcuser 128M Jul  6 20:24 SRR2589044_2.fastq.gz</code></pre>
</div>
<p>There are six FASTQ files ranging from 124M (124MB) to 545M.</p>
</div>
</div>
</div>
</div>
<p>FastQC can accept multiple file names as input, and on both zipped
and unzipped files, so we can use the *.fastq* wildcard to run FastQC on
all of the FASTQ files in this directory.</p>
<div class="codewrapper sourceCode" id="cb18">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="ex">$</span> fastqc <span class="pp">*</span>.fastq<span class="pp">*</span></span></code></pre>
</div>
<p>You will see an automatically updating output message telling you the
progress of the analysis. It will start like this:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Started analysis of SRR2584863_1.fastq
Approx 5% complete for SRR2584863_1.fastq
Approx 10% complete for SRR2584863_1.fastq
Approx 15% complete for SRR2584863_1.fastq
Approx 20% complete for SRR2584863_1.fastq
Approx 25% complete for SRR2584863_1.fastq
Approx 30% complete for SRR2584863_1.fastq
Approx 35% complete for SRR2584863_1.fastq
Approx 40% complete for SRR2584863_1.fastq
Approx 45% complete for SRR2584863_1.fastq</code></pre>
</div>
<p>In total, it should take about five minutes for FastQC to run on all
six of our FASTQ files. When the analysis completes, your prompt will
return. So your screen will look something like this:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Approx 80% complete for SRR2589044_2.fastq.gz
Approx 85% complete for SRR2589044_2.fastq.gz
Approx 90% complete for SRR2589044_2.fastq.gz
Approx 95% complete for SRR2589044_2.fastq.gz
Analysis complete for SRR2589044_2.fastq.gz
$</code></pre>
</div>
<p>The FastQC program has created several new files within our
<code>data/untrimmed_fastq/</code> directory.</p>
<div class="codewrapper sourceCode" id="cb21">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>SRR2584863_1.fastq        SRR2584866_1_fastqc.html  SRR2589044_1_fastqc.html
SRR2584863_1_fastqc.html  SRR2584866_1_fastqc.zip   SRR2589044_1_fastqc.zip
SRR2584863_1_fastqc.zip   SRR2584866_1.fastq.gz     SRR2589044_1.fastq.gz
SRR2584863_2_fastqc.html  SRR2584866_2_fastqc.html  SRR2589044_2_fastqc.html
SRR2584863_2_fastqc.zip   SRR2584866_2_fastqc.zip   SRR2589044_2_fastqc.zip
SRR2584863_2.fastq.gz     SRR2584866_2.fastq.gz     SRR2589044_2.fastq.gz</code></pre>
</div>
<p>For each input FASTQ file, FastQC has created a <code>.zip</code>
file and a</p>
<p><code>.html</code> file. The <code>.zip</code> file extension
indicates that this is actually a compressed set of multiple output
files. We will be working with these output files soon. The
<code>.html</code> file is a stable webpage displaying the summary
report for each of our samples.</p>
<p>We want to keep our data files and our results files separate, so we
will move these output files into a new directory within our
<code>results/</code> directory.</p>
<div class="codewrapper sourceCode" id="cb23">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="ex">$</span> mkdir <span class="at">-p</span> ~/obss_2023/genomic_dna/results/fastqc_untrimmed_reads</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="ex">$</span> mv <span class="pp">*</span>.zip ~/obss_2023/genomic_dna/results/fastqc_untrimmed_reads/</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a><span class="ex">$</span> mv <span class="pp">*</span>.html ~/obss_2023/genomic_dna/results/fastqc_untrimmed_reads/</span></code></pre>
</div>
<p>Now we can navigate into this results directory and do some closer
inspection of our output files.</p>
<div class="codewrapper sourceCode" id="cb24">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="ex">$</span> cd ~/obss_2023/genomic_dna/results/fastqc_untrimmed_reads/</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="viewing-the-fastqc-results">Viewing the FastQC results<a class="anchor" aria-label="anchor" href="#viewing-the-fastqc-results"></a></h3>
<p>We can use the explorer as part of Jupyter hub to navigate to the
directory containing the <code>.html</code> files and click on each one
to view the results of Fastqc.</p>
<div id="original-intstuctions-for-aws" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="original-intstuctions-for-aws" class="callout-inner">
<h3 class="callout-title">Original intstuctions for AWS<a class="anchor" aria-label="anchor" href="#original-intstuctions-for-aws"></a>
</h3>
<div class="callout-content">
<p>If we were working on our local computers, we would be able to look
at each of these HTML files by opening them in a web browser.</p>
<p>However, these files are currently sitting on our remote AWS
instance, where our local computer can not see them. And, since we are
only logging into the AWS instance via the command line - it does not
have any web browser setup to display these files either.</p>
<p>So the easiest way to look at these webpage summary reports will be
to transfer them to our local computers (i.e. your laptop).</p>
<p>To transfer a file from a remote server to our own machines, we will
use <code>scp</code>, which we learned yesterday in the Shell Genomics
lesson.</p>
<p>First we will make a new directory on our computer to store the HTML
files we are transferring. Let’s put it on our desktop for now. Open a
new tab in your terminal program (you can use the pull down menu at the
top of your screen or the Cmd+t keyboard shortcut) and type:</p>
<div class="codewrapper sourceCode" id="cb25">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="ex">$</span> mkdir <span class="at">-p</span> ~/Desktop/fastqc_html</span></code></pre>
</div>
<p>Now we can transfer our HTML files to our local computer using
<code>scp</code>.</p>
<div class="codewrapper sourceCode" id="cb26">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="ex">$</span> scp dcuser@ec2-34-238-162-94.compute-1.amazonaws.com:~/dc_workshop/results/fastqc_untrimmed_reads/<span class="pp">*</span>.html ~/Desktop/fastqc_html</span></code></pre>
</div>
<div id="note-on-using-zsh" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="note-on-using-zsh" class="callout-inner">
<h3 class="callout-title">Note on using zsh<a class="anchor" aria-label="anchor" href="#note-on-using-zsh"></a>
</h3>
<div class="callout-content">
<p>If you are using zsh instead of bash (macOS for example changed the
default recently to zsh), it is likely that a
<code>no matches found</code> error will be displayed. The reason for
this is that the wildcard (“*”) is not correctly interpreted. To fix
this problem the wildcard needs to be escaped with a “\”:</p>
<div class="codewrapper sourceCode" id="cb27">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="ex">$</span> scp dcuser@ec2-34-238-162-94.compute-1.amazonaws.com:~/dc_workshop/results/fastqc_untrimmed_reads/\*.html ~/Desktop/fastqc_html</span></code></pre>
</div>
<p>Alternatively, you can put the whole path into quotation marks:</p>
<div class="codewrapper sourceCode" id="cb28">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a><span class="ex">$</span> scp <span class="st">"dcuser@ec2-34-238-162-94.compute-1.amazonaws.com:~/dc_workshop/results/fastqc_untrimmed_reads/*.html"</span> ~/Desktop/fastqc_html</span></code></pre>
</div>
</div>
</div>
</div>
<p>As a reminder, the first part of the command
<code>dcuser@ec2-34-238-162-94.compute-1.amazonaws.com</code> is the
address for your remote computer. Make sure you replace everything after
<code>dcuser@</code> with your instance number (the one you used to log
in).</p>
<p>The second part starts with a <code>:</code> and then gives the
absolute path of the files you want to transfer from your remote
computer. Do not forget the <code>:</code>. We used a wildcard
(<code>*.html</code>) to indicate that we want all of the HTML
files.</p>
<p>The third part of the command gives the absolute path of the location
you want to put the files. This is on your local computer and is the
directory we just created <code>~/Desktop/fastqc_html</code>.</p>
<p>You should see a status output like this:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>SRR2584863_1_fastqc.html                      100%  249KB 152.3KB/s   00:01
SRR2584863_2_fastqc.html                      100%  254KB 219.8KB/s   00:01
SRR2584866_1_fastqc.html                      100%  254KB 271.8KB/s   00:00
SRR2584866_2_fastqc.html                      100%  251KB 252.8KB/s   00:00
SRR2589044_1_fastqc.html                      100%  249KB 370.1KB/s   00:00
SRR2589044_2_fastqc.html                      100%  251KB 592.2KB/s   00:00</code></pre>
</div>
<p>Now we can go to our new directory and open the 6 HTML files.</p>
<p>Depending on your system, you should be able to select and open them
all at once via a right click menu in your file browser.</p>
</div>
</div>
</div>
<div id="exercise-2" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-2" class="callout-inner">
<h3 class="callout-title">Exercise<a class="anchor" aria-label="anchor" href="#exercise-2"></a>
</h3>
<div class="callout-content">
<p>Discuss your results with a neighbor. Which sample(s) looks the best
in terms of per base sequence quality? Which sample(s) look the
worst?</p>
</div>
</div>
</div>
<div id="accordionSolution3" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution3" aria-expanded="false" aria-controls="collapseSolution3">
  <h4 class="accordion-header" id="headingSolution3">Show me the solution</h4>
</button>
<div id="collapseSolution3" class="accordion-collapse collapse" data-bs-parent="#accordionSolution3" aria-labelledby="headingSolution3">
<div class="accordion-body">
<p>All of the reads contain usable data, but the quality decreases
toward the end of the reads.</p>
</div>
</div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="decoding-the-other-fastqc-outputs">Decoding the other FastQC outputs<a class="anchor" aria-label="anchor" href="#decoding-the-other-fastqc-outputs"></a></h3>
<p>We have now looked at quite a few “Per base sequence quality” FastQC
graphs, but there are nine other graphs that we have not talked about!
Below we have provided a brief overview of interpretations for each of
these plots. For more information, please see the FastQC documentation
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/" class="external-link">here</a></p>
<ul><li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/12%20Per%20Tile%20Sequence%20Quality.html" class="external-link"><strong>Per
tile sequence quality</strong></a>: the machines that perform sequencing
are divided into tiles. This plot displays patterns in base quality
along these tiles. Consistently low scores are often found around the
edges, but hot spots can also occur in the middle if an air bubble was
introduced at some point during the run.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/3%20Per%20Sequence%20Quality%20Scores.html" class="external-link"><strong>Per
sequence quality scores</strong></a>: a density plot of quality for all
reads at all positions. This plot shows what quality scores are most
common.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/4%20Per%20Base%20Sequence%20Content.html" class="external-link"><strong>Per
base sequence content</strong></a>: plots the proportion of each base
position over all of the reads. Typically, we expect to see each base
roughly 25% of the time at each position, but this often fails at the
beginning or end of the read due to quality or adapter content.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/5%20Per%20Sequence%20GC%20Content.html" class="external-link"><strong>Per
sequence GC content</strong></a>: a density plot of average GC content
in each of the reads.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/6%20Per%20Base%20N%20Content.html" class="external-link"><strong>Per
base N content</strong></a>: the percent of times that ‘N’ occurs at a
position in all reads. If there is an increase at a particular position,
this might indicate that something went wrong during sequencing.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/7%20Sequence%20Length%20Distribution.html" class="external-link"><strong>Sequence
Length Distribution</strong></a>: the distribution of sequence lengths
of all reads in the file. If the data is raw, there is often on sharp
peak, however if the reads have been trimmed, there may be a
distribution of shorter lengths.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/8%20Duplicate%20Sequences.html" class="external-link"><strong>Sequence
Duplication Levels</strong></a>: A distribution of duplicated sequences.
In sequencing, we expect most reads to only occur once. If some
sequences are occurring more than once, it might indicate enrichment
bias (e.g. from PCR). If the samples are high coverage (or RNA-seq or
amplicon), this might not be true.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/9%20Overrepresented%20Sequences.html" class="external-link"><strong>Overrepresented
sequences</strong></a>: A list of sequences that occur more frequently
than would be expected by chance.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/10%20Adapter%20Content.html" class="external-link"><strong>Adapter
Content</strong></a>: a graph indicating where adapater sequences occur
in the reads.</li>
<li>
<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/11%20Kmer%20Content.html" class="external-link"><strong>K-mer
Content</strong></a>: a graph showing any sequences which may show a
positional bias within the reads.</li>
</ul></div>
<div class="section level3">
<h3 id="working-with-the-fastqc-text-output">Working with the FastQC text output<a class="anchor" aria-label="anchor" href="#working-with-the-fastqc-text-output"></a></h3>
<p>Now that we have looked at our HTML reports to get a feel for the
data, let’s look more closely at the other output files. Go back to the
tab in your terminal program that is connected to your AWS instance (the
tab label will start with <code>dcuser@ip</code>) and make sure you are
in our results subdirectory.</p>
<div class="codewrapper sourceCode" id="cb30">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="ex">$</span> cd ~/obss_2022/genomic_dna/results/fastqc_untrimmed_reads/</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a><span class="ex">$</span> ls</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>SRR2584863_1_fastqc.html  SRR2584866_1_fastqc.html  SRR2589044_1_fastqc.html
SRR2584863_1_fastqc.zip   SRR2584866_1_fastqc.zip   SRR2589044_1_fastqc.zip
SRR2584863_2_fastqc.html  SRR2584866_2_fastqc.html  SRR2589044_2_fastqc.html
SRR2584863_2_fastqc.zip   SRR2584866_2_fastqc.zip   SRR2589044_2_fastqc.zip</code></pre>
</div>
<p>Our <code>.zip</code> files are compressed files. They each contain
multiple different types of output files for a single input FASTQ file.
To view the contents of a <code>.zip</code> file, we can use the program
<code>unzip</code> to decompress these files. Let’s try doing them all
at once using a wildcard.</p>
<div class="codewrapper sourceCode" id="cb32">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="ex">$</span> unzip <span class="pp">*</span>.zip</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>Archive:  SRR2584863_1_fastqc.zip
caution: filename not matched:  SRR2584863_2_fastqc.zip
caution: filename not matched:  SRR2584866_1_fastqc.zip
caution: filename not matched:  SRR2584866_2_fastqc.zip
caution: filename not matched:  SRR2589044_1_fastqc.zip
caution: filename not matched:  SRR2589044_2_fastqc.zip</code></pre>
</div>
<p>This did not work. We unzipped the first file and then got a warning
message for each of the other <code>.zip</code> files. This is because
<code>unzip</code> expects to get only one zip file as input. We could
go through and unzip each file one at a time, but this is very time
consuming and error-prone. Someday you may have 500 files to unzip!</p>
<p>A more efficient way is to use a <code>for</code> loop like we
learned in the Shell Genomics lesson to iterate through all of our
<code>.zip</code> files. Let’s see what that looks like and then we will
discuss what we are doing with each line of our loop.</p>
<div class="codewrapper sourceCode" id="cb34">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="ex">$</span> for filename in <span class="pp">*</span>.zip</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a><span class="op">&gt;</span> do</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a><span class="op">&gt;</span> unzip <span class="va">$filename</span></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a><span class="op">&gt;</span> done</span></code></pre>
</div>
<p>In this example, the input is six filenames (one filename for each of
our <code>.zip</code> files). Each time the loop iterates, it will
assign a file name to the variable <code>filename</code> and run the
<code>unzip</code> command. The first time through the loop,
<code>$filename</code> is <code>SRR2584863_1_fastqc.zip</code>. The
interpreter runs the command <code>unzip</code> on
<code>SRR2584863_1_fastqc.zip</code>. For the second iteration,
<code>$filename</code> becomes <code>SRR2584863_2_fastqc.zip</code>.
This time, the shell runs <code>unzip</code> on
<code>SRR2584863_2_fastqc.zip</code>. It then repeats this process for
the four other <code>.zip</code> files in our directory.</p>
<p>When we run our <code>for</code> loop, you will see output that
starts like this:</p>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code><span><span class="va">Archive</span><span class="op">:</span>  <span class="va">SRR2589044_2_fastqc.zip</span></span>
<span>   <span class="va">creating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span></span>
<span>   <span class="va">creating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Icons</span><span class="op">/</span></span>
<span>   <span class="va">creating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Icons</span><span class="op">/</span><span class="va">fastqc_icon.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Icons</span><span class="op">/</span><span class="va">warning.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Icons</span><span class="op">/</span><span class="va">error.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Icons</span><span class="op">/</span><span class="va">tick.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">summary.txt</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">per_base_quality.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">per_tile_quality.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">per_sequence_quality.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">per_base_sequence_content.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">per_sequence_gc_content.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">per_base_n_content.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">sequence_length_distribution.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">duplication_levels.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">Images</span><span class="op">/</span><span class="va">adapter_content.png</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">fastqc_report.html</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">fastqc_data.txt</span></span>
<span>  <span class="va">inflating</span><span class="op">:</span> <span class="va">SRR2589044_2_fastqc</span><span class="op">/</span><span class="va">fastqc.fo</span></span></code></pre>
</div>
<p>The <code>unzip</code> program is decompressing the <code>.zip</code>
files and creating a new directory (with subdirectories) for each of our
samples, to store all of the different output that is produced by
FastQC. There</p>
<p>are a lot of files here. The one we are going to focus on is the
<code>summary.txt</code> file.</p>
<p>If you list the files in our directory now you will see:</p>
<pre><code>SRR2584863_1_fastqc       SRR2584866_1_fastqc       SRR2589044_1_fastqc
SRR2584863_1_fastqc.html  SRR2584866_1_fastqc.html  SRR2589044_1_fastqc.html
SRR2584863_1_fastqc.zip   SRR2584866_1_fastqc.zip   SRR2589044_1_fastqc.zip
SRR2584863_2_fastqc       SRR2584866_2_fastqc       SRR2589044_2_fastqc
SRR2584863_2_fastqc.html  SRR2584866_2_fastqc.html  SRR2589044_2_fastqc.html
SRR2584863_2_fastqc.zip   SRR2584866_2_fastqc.zip   SRR2589044_2_fastqc.zip</code></pre>
<p>The <code>.html</code> files and the uncompressed <code>.zip</code>
files are still present, but now we also have a new directory for each
of our samples. We can see for sure that it is a directory if we use the
<code>-F</code> flag for <code>ls</code>.</p>
<div class="codewrapper sourceCode" id="cb37">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span></span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>SRR2584863_1_fastqc/      SRR2584866_1_fastqc/      SRR2589044_1_fastqc/
SRR2584863_1_fastqc.html  SRR2584866_1_fastqc.html  SRR2589044_1_fastqc.html
SRR2584863_1_fastqc.zip   SRR2584866_1_fastqc.zip   SRR2589044_1_fastqc.zip
SRR2584863_2_fastqc/      SRR2584866_2_fastqc/      SRR2589044_2_fastqc/
SRR2584863_2_fastqc.html  SRR2584866_2_fastqc.html  SRR2589044_2_fastqc.html
SRR2584863_2_fastqc.zip   SRR2584866_2_fastqc.zip   SRR2589044_2_fastqc.zip</code></pre>
</div>
<p>Let’s see what files are present within one of these output
directories.</p>
<div class="codewrapper sourceCode" id="cb39">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="ex">$</span> ls <span class="at">-F</span> SRR2584863_1_fastqc/</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>fastqc_data.txt  fastqc.fo  fastqc_report.html	Icons/	Images/  summary.txt</code></pre>
</div>
<p>Use <code>less</code> to preview the <code>summary.txt</code> file
for this sample.</p>
<div class="codewrapper sourceCode" id="cb41">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="ex">$</span> less SRR2584863_1_fastqc/summary.txt</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>PASS    Basic Statistics        SRR2584863_1.fastq
PASS    Per base sequence quality       SRR2584863_1.fastq
PASS    Per tile sequence quality       SRR2584863_1.fastq
PASS    Per sequence quality scores     SRR2584863_1.fastq
WARN    Per base sequence content       SRR2584863_1.fastq
WARN    Per sequence GC content SRR2584863_1.fastq
PASS    Per base N content      SRR2584863_1.fastq
PASS    Sequence Length Distribution    SRR2584863_1.fastq
PASS    Sequence Duplication Levels     SRR2584863_1.fastq
PASS    Overrepresented sequences       SRR2584863_1.fastq
WARN    Adapter Content SRR2584863_1.fastq</code></pre>
</div>
<p>The summary file gives us a list of tests that FastQC ran, and tells
us whether this sample passed, failed, or is borderline
(<code>WARN</code>). Remember, to quit from <code>less</code> you must
type <code>q</code>.</p>
</div>
<div class="section level3">
<h3 id="documenting-our-work">Documenting our work<a class="anchor" aria-label="anchor" href="#documenting-our-work"></a></h3>
<p>We can make a record of the results we obtained for all our
samples</p>
<p>by concatenating all of our <code>summary.txt</code> files into a
single file using the <code>cat</code> command. We will call this
<code>fastqc_summaries.txt</code> and move it to
<code>~/obss_2023/genomic_dna/docs</code>.</p>
<div class="codewrapper sourceCode" id="cb43">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb43-1"><a href="#cb43-1" tabindex="-1"></a><span class="ex">$</span> cat <span class="pp">*</span>/summary.txt <span class="op">&gt;</span> ~/obss_2023/genomic_dna/docs/fastqc_summaries.txt</span></code></pre>
</div>
<div id="exercise-3" class="callout challenge">
<div class="callout-square">
<i class="callout-icon" data-feather="zap"></i>
</div>
<div id="exercise-3" class="callout-inner">
<h3 class="callout-title">Exercise<a class="anchor" aria-label="anchor" href="#exercise-3"></a>
</h3>
<div class="callout-content">
<p>Which samples failed at least one of FastQC’s quality tests? What
test(s) did those samples fail?</p>
</div>
</div>
</div>
<div id="accordionSolution4" class="accordion challenge-accordion accordion-flush">
<div class="accordion-item">
<button class="accordion-button solution-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#collapseSolution4" aria-expanded="false" aria-controls="collapseSolution4">
  <h4 class="accordion-header" id="headingSolution4">Show me the solution</h4>
</button>
<div id="collapseSolution4" class="accordion-collapse collapse" data-bs-parent="#accordionSolution4" aria-labelledby="headingSolution4">
<div class="accordion-body">
<p>We can get the list of all failed tests using <code>grep</code>.</p>
<div class="codewrapper sourceCode" id="cb44">
<h3 class="code-label">BASH<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="sourceCode bash" tabindex="0"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="ex">$</span> cd ~/obss_2023/genomic_dna/docs</span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a><span class="ex">$</span> grep FAIL fastqc_summaries.txt</span></code></pre>
</div>
<div class="codewrapper">
<h3 class="code-label">OUTPUT<i aria-hidden="true" data-feather="chevron-left"></i><i aria-hidden="true" data-feather="chevron-right"></i>
</h3>
<pre class="output" tabindex="0"><code>FAIL    Per base sequence quality       SRR2584863_2.fastq.gz
FAIL    Per tile sequence quality       SRR2584863_2.fastq.gz
FAIL    Per base sequence content       SRR2584863_2.fastq.gz
FAIL    Per base sequence quality       SRR2584866_1.fastq.gz
FAIL    Per base sequence content       SRR2584866_1.fastq.gz
FAIL    Adapter Content SRR2584866_1.fastq.gz
FAIL    Adapter Content SRR2584866_2.fastq.gz
FAIL    Adapter Content SRR2589044_1.fastq.gz
FAIL    Per base sequence quality       SRR2589044_2.fastq.gz
FAIL    Per tile sequence quality       SRR2589044_2.fastq.gz
FAIL    Per base sequence content       SRR2589044_2.fastq.gz
FAIL    Adapter Content SRR2589044_2.fastq.gz</code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</section><section id="other-notes-optional"><h2 class="section-heading">Other notes – optional<a class="anchor" aria-label="anchor" href="#other-notes-optional"></a>
</h2>
<hr class="half-width"><div id="quality-encodings-vary" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="quality-encodings-vary" class="callout-inner">
<h3 class="callout-title">Quality encodings vary<a class="anchor" aria-label="anchor" href="#quality-encodings-vary"></a>
</h3>
<div class="callout-content">
<p>Although we have used a particular quality encoding system to
demonstrate interpretation of read quality, different sequencing
machines use different encoding systems. This means that, depending on
which sequencer you use to generate your data, a <code>#</code> may not
be an indicator of a poor quality base call.</p>
<p>This mainly relates to older Solexa/Illumina data, but it is
essential that you know which sequencing platform was used to generate
your data, so that you can tell your quality control program which
encoding to use. If you choose the wrong encoding, you run the risk of
throwing away good reads or (even worse) not throwing away bad
reads!</p>
</div>
</div>
</div>
<div id="same-symbols-different-meanings" class="callout">
<div class="callout-square">
<i class="callout-icon" data-feather="bell"></i>
</div>
<div id="same-symbols-different-meanings" class="callout-inner">
<h3 class="callout-title">Same symbols, different meanings<a class="anchor" aria-label="anchor" href="#same-symbols-different-meanings"></a>
</h3>
<div class="callout-content">
<p>Here we see <code>&gt;</code> being used as a shell prompt, whereas
<code>&gt;</code> is also used to redirect output. Similarly,
<code>$</code> is used as a shell prompt, but, as we saw earlier, it is
also used to ask the shell to get the value of a variable.</p>
<p>If the <em>shell</em> prints <code>&gt;</code> or <code>$</code> then
it expects you to type something, and the symbol is a prompt.</p>
<p>If <em>you</em> type <code>&gt;</code> or <code>$</code> yourself, it
is an instruction from you that the shell should redirect output or get
the value of a variable.</p>
</div>
</div>
</div>
<div id="keypoints1" class="callout keypoints">
<div class="callout-square">
<i class="callout-icon" data-feather="key"></i>
</div>
<div class="callout-inner">
<h3 class="callout-title">Key Points<a class="anchor" aria-label="anchor" href="#keypoints1"></a>
</h3>
<div class="callout-content">
<ul><li>Quality encodings vary across sequencing platforms.</li>
<li>
<code>for</code> loops let you perform the same set of operations on
multiple files with a single command.</li>
</ul></div>
</div>
</div>
</section></div> <!-- / div.lesson-content -->
    </main><!-- / main#main-content.main-content --><nav class="bottom-pagination mx-md-4" aria-label="Previous and Next Chapter"><div class="d-block d-sm-block d-md-none">
        <a class="chapter-link" href="../instructor/01-background.html"><i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>Previous</a>
        <a class="chapter-link float-end" href="../instructor/03-trimming.html">Next<i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i></a>
      </div>
      <!-- content for large screens -->
      <div class="d-none d-sm-none d-md-block">
        <a class="chapter-link" href="../instructor/01-background.html" rel="prev">
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-left"></i>
          Previous: Background and
        </a>
        <a class="chapter-link float-end" href="../instructor/03-trimming.html" rel="next">
          Next: Trimming and... 
          <i aria-hidden="true" class="small-arrow" data-feather="arrow-right"></i>
        </a>
      </div>
    </nav></div> <!-- / div.primary-content.col-xs-12 -->
<!-- END:   inst/pkgdown/templates/content-instructor.html-->

      </div><!--/div.row-->
      		<footer class="row footer mx-md-3"><hr><div class="col-md-6">
        <p>This lesson is subject to the <a href="CODE_OF_CONDUCT.html">Code of Conduct</a></p>
        <p>
        
        <a href="https://github.com/otagobioinformaticsspringschool/wrangling-genomics-nesi/edit/main/episodes/02-quality-control.md" class="external-link">Edit on GitHub</a>
        
	
        | <a href="https://github.com/otagobioinformaticsspringschool/wrangling-genomics-nesi/blob/main/CONTRIBUTING.md" class="external-link">Contributing</a>
        | <a href="https://github.com/otagobioinformaticsspringschool/wrangling-genomics-nesi/" class="external-link">Source</a></p>
				<p><a href="https://github.com/otagobioinformaticsspringschool/wrangling-genomics-nesi/blob/main/CITATION" class="external-link">Cite</a> | <a href="mailto:team@carpentries.org">Contact</a> | <a href="https://carpentries.org/about/" class="external-link">About</a></p>
			</div>
			<div class="col-md-6">
        
        <p>Materials licensed under <a href="LICENSE.html">CC-BY 4.0</a> by the authors</p>
        
        <p>Template licensed under <a href="https://creativecommons.org/licenses/by-sa/4.0/" class="external-link">CC-BY 4.0</a> by <a href="https://carpentries.org/" class="external-link">The Carpentries</a></p>
        <p>Built with <a href="https://github.com/carpentries/sandpaper/tree/0.16.3" class="external-link">sandpaper (0.16.3)</a>, <a href="https://github.com/carpentries/pegboard/tree/0.7.4" class="external-link">pegboard (0.7.4)</a>, and <a href="https://github.com/carpentries/varnish/tree/1.0.1" class="external-link">varnish (1.0.1)</a></p>
			</div>
		</footer></div> <!-- / div.container -->
	<div id="to-top">
		<a href="#top">
      <i class="search-icon" data-feather="arrow-up" role="img" aria-label="Back To Top"></i><br><!-- <span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top --><span class="d-none d-sm-none d-md-none d-lg-none d-xl-block">Back</span> To Top
		</a>
	</div>
  <script type="application/ld+json">
    {
  "@context": "https://schema.org",
  "@type": "TrainingMaterial",
  "@id": "https://otagobioinformaticsspringschool.github.io/wrangling-genomics-nesi/instructor/02-quality-control.html",
  "inLanguage": "en",
  "dct:conformsTo": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
  "description": "A Carpentries Lesson teaching foundational data and coding skills to researchers worldwide",
  "keywords": "software, data, lesson, The Carpentries",
  "name": "Assessing Read Quality",
  "creativeWorkStatus": "active",
  "url": "https://otagobioinformaticsspringschool.github.io/wrangling-genomics-nesi/instructor/02-quality-control.html",
  "identifier": "https://otagobioinformaticsspringschool.github.io/wrangling-genomics-nesi/instructor/02-quality-control.html",
  "dateCreated": "2015-03-24",
  "dateModified": "2023-11-27",
  "datePublished": "2024-04-09"
}

  </script><script>
		feather.replace();
	</script><!-- Matomo
    2022-11-07: we have gotten a notification that we have an overage for our
    tracking and I'm pretty sure this has to do with Workbench usage.
    Considering that I am not _currently_ using this tracking because I do not
    yet know how to access the data, I am turning this off for now.
  <script>
    var _paq = window._paq = window._paq || [];
    /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
    _paq.push(["setDocumentTitle", document.domain + "/" + document.title]);
    _paq.push(["setDomains", ["*.preview.carpentries.org","*.datacarpentry.github.io","*.datacarpentry.org","*.librarycarpentry.github.io","*.librarycarpentry.org","*.swcarpentry.github.io", "*.carpentries.github.io"]]);
    _paq.push(["setDoNotTrack", true]);
    _paq.push(["disableCookies"]);
    _paq.push(['trackPageView']);
    _paq.push(['enableLinkTracking']);
    (function() {
          var u="https://carpentries.matomo.cloud/";
          _paq.push(['setTrackerUrl', u+'matomo.php']);
          _paq.push(['setSiteId', '1']);
          var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
          g.async=true; g.src='https://cdn.matomo.cloud/carpentries.matomo.cloud/matomo.js'; s.parentNode.insertBefore(g,s);
        })();
  </script>
  End Matomo Code --></body></html><!-- END:   inst/pkgdown/templates/layout.html-->

